### DISCUSSION TITLE
Adding Text To Existing Image,How should I optimize my code?

### DISCUSSION THREAD

**Discussion by xiranyu** [dlemstra/Magick.NET#1124] [Help]
This code took 1000ms

           using (var image = new MagickImage(sourceImgPath))
            {
                using (var caption = new MagickImage($"caption:{text}", readSettings))
                {
                    image.Composite(caption, x1, y1, CompositeOperator.Over);
                }

                image.Write(newImgPath);
            }

This code took 2000ms

           using (var image = new MagickImage(sourceImgPath))
            {
                using (var caption = new MagickImage($"caption:{text}", readSettings))
                {
                    image.Composite(caption, x1, y1, CompositeOperator.Over);
                }

                using (var caption = new MagickImage($"caption:{text2}", readSettings))
                {
                    image.Composite(caption, x2, y2, CompositeOperator.Over);
                }

                image.Write(newImgPath);
            }

Is there any way to optimize it? thanks.

**Comment by dlemstra** [comment]
The drawing code is not very performant because it does a lot of checks to get the best output. I don't think you can do anything to optimize this. You are probably drawing a lot of information on the image.
